xbmc/interfaces/json-rpc/PVROperations.cpp

```diff
@@ -145,6 +146,25 @@ JSONRPC_STATUS CPVROperations::GetChannelDetails(const std::string &method, ITra
   return OK;
 }
 
+JSONRPC_STATUS CPVROperations::GetClients(const std::string& method,
+                                          ITransportLayer* transport,
+                                          IClient* client,
+                                          const CVariant& parameterObject,
+                                          CVariant& result)
+{
+  if (!CServiceBroker::GetPVRManager().IsStarted())
+    return FailedToExecute;
+
+  int start, end;
+
+  auto clientInfos = CServiceBroker::GetPVRManager().Clients()->GetEnabledClientInfos();
+  HandleLimits(parameterObject, result, clientInfos.size(), start, end);
+  for (int index = start; index < end; index++)
+    result["clients"].append(clientInfos[index]);
```

Comment:
no need for an index.

```c++
  for (const auto& clientInfo : clientInfos)
    result["clients"].append(clientInfo);
```

Reply:
`HandleLimits` is for paging? This does not really work. You can't ensure that between to calls to GetEnabledClientInfos() the amount and/or order of clients dos not change.